### PR TITLE
fix(pie chart): add kws consumption to the Pie Chart by (Category/Meters)

### PR DESCRIPTION
**Pull request by ceoaliongroo**
#383

screen shot 2015-06-11 at 6 22 14 pm

screen shot 2015-06-14 at 5 17 35 pm

### CODE REVIEW

**Comment by ceoaliongroo** [comment]
@amitaibu ready to code review

File: client/app/scripts/controllers/details.js [hunk 1]
```diff
@@ -1,19 +1,40 @@
 'use strict';
 
 angular.module('negawattClientApp')
-  .controller('DetailsCtrl', function ($scope, $state, $stateParams, ChartCategories, categoriesChart, meters) {
-    var categoryId;
+  .controller('DetailsCtrl', function ($scope, $state, $stateParams, FilterFactory, ChartCategories, meters, $filter, categories) {
+    var categoryId,
```
Comment (amitaibu) [inline client/app/scripts/controllers/details.js]: doc

File: client/app/scripts/controllers/details.js [hunk 2]
```diff
@@ -1,19 +1,40 @@
...
 
     // Select category form the pie chart.
     $scope.onSelect = function(selectedItem, chartData) {
-      categoryId = ChartCategories.getCategoryIdSelected(selectedItem, chartData);
-      if (angular.isDefined(categoryId)) {
-        $state.go('dashboard.withAccount.categories', {accountId: $stateParams.accountId, categoryId: categoryId, chartNextPeriod: undefined, chartPreviousPeriod: undefined});
+      if (summary.type === 'category') {
+        categoryId = ChartCategories.getCategoryIdSelected(selectedItem, chartData);
+        if (angular.isDefined(categoryId)) {
+          $state.go('dashboard.withAccount.categories', {accountId: $stateParams.accountId, categoryId: categoryId, chartNextPeriod: undefined, chartPreviousPeriod: undefined});
```
Comment (amitaibu) [inline client/app/scripts/controllers/details.js]: check if need the `undefnied`

File: client/app/scripts/controllers/details.js [hunk 3]
```diff
@@ -1,19 +1,40 @@
...
+    // Handle lazy-load of electricity data.
+    // When cache expands, update the chart.
+    $scope.$on("nwElectricityChanged", function(event, electricity) {
+      summary = $filter('summary')(electricity[FilterFactory.get('activeElectricityHash')]);
+
+      if (angular.isUndefined(summary)) {
+        return;
+      }
+      // Update electricity object.
+      $scope.categoriesChart = $filter('toPieChartDataset')(summary, (summary.type === 'category') ? $scope.categories.collection : $scope.meters.listAll);
```
Comment (amitaibu) [inline client/app/scripts/controllers/details.js]: split to new variable + docs

File: client/app/scripts/controllers/details.js [hunk 4]
```diff
@@ -1,19 +1,40 @@
...
+
+      event.preventDefault();
```
Comment (amitaibu) [inline client/app/scripts/controllers/details.js]: remove

**Comment by ceoaliongroo** [comment]
Merged.

**Pull request closed without merging**